### PR TITLE
Stream Response

### PR DESCRIPTION
**Pull request by benrudolph**
@wpride @snopoke this is some work towards only dealing with restores via streams. not completely finished but wanted to get opinions on the approach. so far this correctly saves all the restore xml into postgres and you can open a form. i think there's an error occurring when answering a question where it tries to read the full stream twice. i don't think loading the entire restore into memory is tenable for enikshay if we plan on having 100s of concurrent users

### CODE REVIEW

File: src/main/java/repo/impl/PostgresFormSessionRepo.java [hunk 1]
```diff
@@ -65,23 +65,85 @@
...
+        public void setValues(PreparedStatement ps) throws SQLException {
+            byte[] sessionDataBytes = writeToBytes(mSession.getSessionData());
+
+            ps.setString(1, mSession.getInstanceXml());
+            ps.setString(2, mSession.getId());
+            ps.setString(3, mSession.getFormXml());
+            try {
+                ps.setCharacterStream(4, new InputStreamReader(mSession.getRestoreXml(), "UTF-8"), mSession.getRestoreXml().available());
+            } catch (UnsupportedEncodingException e) {
+                e.printStackTrace();
```
Comment (benrudolph) [inline src/main/java/repo/impl/PostgresFormSessionRepo.java]: will fix all these in another pass

File: src/main/java/hq/CaseAPIs.java [hunk 2]
```diff
@@ -41,12 +42,12 @@ public static UserSqlSandbox restoreIfNotExists(RestoreFactory restoreFactory, b
             return restoreFactory.getSqlSandbox();
         } else{
             db.getParentFile().mkdirs();
-            String xml = restoreFactory.getRestoreXml(overwriteCache);
-            return restoreUser(restoreFactory.getWrappedUsername(), restoreFactory.getDbPath(), xml);
+            InputStream restoreStream = restoreFactory.getRestoreXml(overwriteCache);
```
Comment (benrudolph) [inline src/main/java/hq/CaseAPIs.java]: 😂

**Comment by benrudolph** [comment]
going to also have to figure out a strategy for when we want to get a new stream handler, i'm thinking if the stream is exhausted we fetch a new stream from the one saved in postgres

File: src/main/java/services/RestoreFactory.java [hunk 3]
```diff
@@ -64,7 +67,7 @@
 
     private final Log log = LogFactory.getLog(RestoreFactory.class);
 
-    private String cachedRestore = null;
+    private InputStream cachedRestore = null;
```
Comment (benrudolph) [inline src/main/java/services/RestoreFactory.java]: we probably won't need this anymore

**Comment by snopoke** [comment]
Don't have that much context on this. What's the reason for the change?

**Comment by benrudolph** [comment]
We're currently thinking about improving restores on web apps for many users, one low hanging fruit i thought was changing the code to use streams instead of loading the entire payload into memory. their smallest load test user has a 17mb restore payload. i had actually imagined that their restore payload was larger, so maybe this change isn't worth it as much

**Pull request closed without merging**